server/src/api/api.ts

```diff
@@ -170,9 +169,85 @@ const resolvers: any = {
 
         return {
             id: newObjId[0],
+            category: savedCategory,
             ...args.newItem
         };
     },
+    /**
+     * Update an existing item given its ID
+     * TODO: reduce duplicate code from createItem
+     * @param root
+     * @param _args
+     * @param _context
+     */
+    updateItem: async (root, _args, _context) => {
+        // @ts-ignore
+        const {args, context} = fixArguments(root, _args, _context);
+
+        // Restrict endpoint to admins
+        if (!context.user.admin) {
+            throw new GraphQLError("You do not have permission to access the updateItem endpoint");
+        }
+
+        if (!args.id || args.id <= 0) {
+            throw new GraphQLError("You must provide a valid item ID (greater than or equal to 0) to update an item.");
+        }
+
+        if (!args.updatedItem.item_name.trim().length) {
+            throw new GraphQLError("The item name (item_name) can't be empty.");
+        }
+
+        if (!args.updatedItem.category.trim().length) {
+            throw new GraphQLError("The category for this item can't be blank.");
+        }
+
+        if (args.updatedItem.totalAvailable < 0) {
+            throw new GraphQLError(`The total quantity available (totalQtyAvailable) for a new item can't be less than 0.  Value provided: ${args.updatedItem.totalAvailable}`);
+        }
+
+        if (args.updatedItem.maxRequestQty < 1) {
+            throw new GraphQLError(`The max request quantity (maxRequestQty) must be at least 1.  Value provided: ${args.updatedItem.maxRequestQty}`);
```

Comment:
what catches these errors, curious?

Reply:
It shows up in your GraphQL client.  Ideally client-side validation will stop you before you ever see this